### PR TITLE
Enable non-zero-exit in case of an error while running "build"

### PR DESCRIPTION
**Pull request by nenadilic84**
## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
Resolves the issue: #183

## How Has This Been Tested?

```
$ ./faas-cli build
Error: please provide a valid -image name for your Docker image
# checking the exit code
$ echo $?
1
```
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by nenadilic84** [comment]
@alexellis, not sure if I know what you mean by 

> I'd like to see the errors triggered in the comments here.

**Comment by alexellis** [comment]
You tested one of the error conditions. I'd like you to run each error scenario and show that it triggers the exit code we want. @itscaro may also be able to help show to extend the tests to pick up the exit code automatically.

File: commands/build.go [hunk 1]
```diff
@@ -80,27 +79,26 @@ func runBuild(cmd *cobra.Command, args []string) {
 	}
 
 	if pullErr := PullTemplates(""); pullErr != nil {
-		log.Fatalln("Could not pull templates for OpenFaaS.", pullErr)
+		log.Fatalf("Could not pull templates for OpenFaaS.")
+		return pullErr
```
Comment (alexellis) [inline commands/build.go]: log.Fatalf will probably cause an exit here. Let's use a combined error message with fmt.Errorf of pullErr and our custom message and return it as the error.

File: commands/faas.go [hunk 2]
```diff
@@ -44,7 +45,9 @@ func init() {
 func Execute(customArgs []string) {
 	faasCmd.SilenceUsage = true
 	faasCmd.SetArgs(customArgs[1:])
-	faasCmd.Execute()
+	if err := faasCmd.Execute(); err != nil {
```
Comment (alexellis) [inline commands/faas.go]: Nice

**Comment by alexellis** [comment]
Pinging @itscaro to suggest how to test the exit code from the CLI integration tests.

**Comment by nenadilic84** [comment]
Will push the changes probably till the end of the day...

**Comment by nenadilic84** [comment]
@alexellis, update the PR and added some tests... Let me know if you have something else in mind

**Comment by alexellis** [comment]
I think we can merge this now. Not 100% on the tests. I think it's hard to test but we only have the negative cases not a positive one. Ideally want both.

**Comment by nenadilic84** [comment]
@alexellis, not sure what you mean? Should I add more tests for the `build` command?

**Comment by alexellis** [comment]
There is only a test that shows the negative case (exit code non-zero) - we need a test to show positive case of (exit code zero).

**Comment by alexellis** [comment]
Merging work so far, thank you 👍
